fe/fe-core/src/main/java/org/apache/doris/nereids/memo/Memo.java

```diff
@@ -169,7 +169,8 @@ private Pair<Boolean, GroupExpression> rewriteGroupExpression(
             GroupExpression groupExpression, Group target, LogicalProperties logicalProperties) {
         boolean newGroupExpressionGenerated = true;
         GroupExpression existedGroupExpression = groupExpressions.get(groupExpression);
-        if (existedGroupExpression != null) {
+        if (existedGroupExpression != null
+                && (target == null || target.equals(existedGroupExpression.getOwnerGroup()))) {
```

Comment:
if target is not null, and not the owner group of existedGroupExpression, should we merge them, as we do in insertGroupExpression()?
Otherwise, there are two groups with equivalent group expressions in Memo.
I think this issue is introduced by former pr, which split insertGroupExpression() and rewriteGroupExpression() from insertOrRewriteGroupExpression().

Reply:
looks like todo. we can merge this pr and complete repair it later